examples/dataflow-data-generator/data-generator-pipeline/data_generator_pipeline.py

```diff
@@ -0,0 +1,150 @@
+from a json file and writes random data of the schema's shape to a 
+BigQuery table or as CSV or AVRO files on GCS. This can be used to 
+ease apprehension about BQ costs, unblock integration testing before
+real data can be provided by the business, or create dummy datasets 
+for stress testing in the event of large data growth.
+"""
+
+from __future__ import absolute_import
+import json
+import logging
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+from data_generator.DataGenerator import DataGenerator, FakeRowGen, \
+    parse_data_generator_args, validate_data_args, fetch_schema,\
+    write_n_line_file_to_gcs
+import avro.schema
+
+from data_generator.CsvUtil import dict_to_csv
+from data_generator.AvroUtil import fix_record_for_avro
+from data_generator.enforce_primary_keys import EnforcePrimaryKeys
+
+def run(argv=None):
+    """
+    This funciton parses the command line arguments and runs the Beam Pipeline.
+
+    Args:
+        argv: list containing the commandline arguments for this call of the
+         script.
+    """
+    # Keeps track if schema was inferred by input or ouput table.
+    schema_inferred = False
+
+    data_args, pipeline_args = parse_data_generator_args(argv)
+    data_args, schema_inferred = fetch_schema(data_args, schema_inferred)
+    pipeline_options = PipelineOptions(pipeline_args)
+
+    temp_location = pipeline_options.display_data()['temp_location']
+    temp_blob = write_n_line_file_to_gcs(
+        pipeline_options.display_data()['project'],
+        temp_location,
+        data_args.num_records)
+
+    data_gen = DataGenerator(bq_schema_filename=data_args.schema_file,
+                             input_bq_table=data_args.input_bq_table,
+                             p_null=data_args.p_null,
+                             n_keys=data_args.n_keys,
+                             min_date=data_args.min_date,
+                             max_date=data_args.max_date,
+                             only_pos=data_args.only_pos,
+                             max_int=data_args.max_int,
+                             max_float=data_args.max_float,
+                             float_precision=data_args.float_precision,
+                             write_disp=data_args.write_disp,
+                             key_skew=data_args.key_skew,
+                             primary_key_cols=data_args.primary_key_cols)
+
+
+    # Initiate the pipeline using the pipeline arguments passed in from the
+    # command line.  This includes information including where Dataflow should
+    # store temp files, and what the project id is and what runner to use.
+    p = beam.Pipeline(options=pipeline_options)
+
+    rows = (p
+        # Read the file we created with num_records newlines.
+        | 'Read file with num_records lines' >> beam.io.ReadFromText(
+                '/'.join(['gs:/', temp_blob.bucket.name, temp_blob.name])
```

Comment:
Can this line be os.path.join('gs://',"temp_blob.bucket", "temp_blob.name")? Provided we do not have / in the beginning of "temp_blob.bucket", "temp_blob.name"?

Reply:
changed to your suggestion.